elixir/apps/api/lib/api/controllers/resource_controller.ex

```diff
@@ -0,0 +1,55 @@
+defmodule API.ResourceController do
+  use API, :controller
+  import API.ControllerHelpers
+  alias Domain.Resources
+
+  action_fallback API.FallbackController
+
+  def index(conn, params) do
+    list_opts = params_to_list_opts(params)
+
+    with {:ok, resources, metadata} <-
+           Resources.list_resources(conn.assigns.subject, list_opts) do
+      render(conn, :index, resources: resources, metadata: metadata)
+    end
+  end
+
+  def show(conn, %{"id" => id}) do
+    with {:ok, resource} <- Resources.fetch_resource_by_id(id, conn.assigns.subject) do
+      render(conn, :show, resource: resource)
+    end
+  end
+
+  def create(conn, %{"resource" => params}) do
+    with attrs <- set_param_defaults(params),
```

Comment:
You don't want to put `=` as the first element of with, that will generate an `case` clause like this:
```
case set_param_defaults(params) do
  attrs -> ..rest of with block..
end
```
```suggestion
    attrs = set_param_defaults(params)
    
    with 
```

Reply:
I made this change and I can see why in this case it wouldn't be necessary to have that in the `with` clause, but I'm slightly confused about the `with` clause generating a `case` clause.  I'm guessing I just need to read up on `with` a big more.  Is there a specific set of docs talking about this that I could read?

Reply:
I also wanted to clarify something for my own edification. You mentioned not having a `=` in the first element of a `with` statement.  The original code I had used a `<-`, is that the same as using a `=` and if so, how is that different from something like:
```
with {:ok, foo} <- do_something() do
  ...block of code here...
end
```

Reply:
@bmanifold the `with` clause just compiles down to a bunch of `case`s:
```
with :foo <- foo(),
        :bar <- bar() do
  :ok
else
  :fiz -> {:error, :fiz11}
  :fix -> {:error, :fix}
end
```
is the same as:
```
case foo() do
   :foo -> 
   case bar() do
       :bar -> :ok
      :fiz -> {:error, :fiz11}
      :fix -> {:error, :fix}
    end
  :fiz -> {:error, :fiz11}
  :fix -> {:error, :fix}
end
```

So:
```
with foo <- foo() do
...code
end
=>
case foo do 
   foo -> ...code
end
=> 
...code
```

Reply:
This makes sense.  I really appreciate the detailed explanation 🙏